src/ASM/ASMs3D.C

```diff
@@ -3041,7 +3042,8 @@ void ASMs3D::generateThreadGroups (size_t strip1, size_t strip2, size_t strip3,
       // Verify that the nodes on this thread are not present on the others
       for (k = 0; k < j; k++)
         for (nit = nodes[j].begin(); nit != nodes[j].end(); ++nit)
-          if (nodes[k].find(*nit) != nodes[k].end())
+          if ((!this->isLMn(*nit+1) || this->getLMType(*nit+1) != 'G' || !ignoreGlobalLM) &&
+              nodes[k].find(*nit) != nodes[k].end())
             std::cout <<"\n  ** Warning: Node "<< *nit <<" is present on both"
```

Comment:
Samme kommentar som for ASMs2D. Her har du forresten ikke lagt til bracees, så da så..